samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java

```diff
@@ -60,16 +74,22 @@
   private final Config config;
   private final boolean fetchThresholdBytesEnabled;
   private final KafkaSystemConsumerMetrics metrics;
+  private final KafkaStartpointRegistrationHandler kafkaStartpointRegistrationHandler;
 
   // This sink is used to transfer the messages from the proxy/consumer to the BlockingEnvelopeMap.
   final KafkaConsumerMessageSink messageSink;
 
   // This proxy contains a separate thread, which reads kafka messages (with consumer.poll()) and populates
   // BlockingEnvelopMap's buffers.
-  final private KafkaConsumerProxy proxy;
+  @VisibleForTesting
+  KafkaConsumerProxy proxy;
 
   // keep registration data until the start - mapping between registered SSPs and topicPartitions, and their offsets
   final Map<TopicPartition, String> topicPartitionsToOffset = new HashMap<>();
```

Comment:
2A. I agree that it is odd for specific `SystemConsumer` implementations to each compare offsets. It seems outside of the scope of this PR though. Can we move that comparator logic refactoring to a separate PR so that we can cleanly separate the changes? That PR could be committed before this PR.
2B. I looked at some other code which uses `register` (e.g. `ContainerStorageManager`, `CoordinatorStreamStore`, and `CoordinatorStreamSystemConsumer`). Those don't really seem to do any comparisons to make sure they are registering the lowest offset. How can you tell all of the upper-layers already pass the lowest offset? I just want to double check that there wasn't a bug in an upper-layer which happened to be prevented by having offset comparator logic directly in `SystemConsumer` impls.

Reply:
1. Created PR: 954 which addresses the comment of moving the refactor into separate PR. 
2. Added the rationale in description of PR: 954 as to why this refactor would not break any existing functionality.
    * `CoordinatorStreamSystemConsumer`: Reads the coordinator stream from oldest offset. Please check here(https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/coordinator/stream/CoordinatorStreamSystemConsumer.java#L128) for more details.
    * `CoordinatorStreamStore`: Reads the coordinator stream from oldest offset. Please check [here](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/coordinator/metadatastore/CoordinatorStreamStore.java#L183) for more details. 
    *  `ContainerStorageManager`: Chooses the lowest offset before registering with consumer. [Relevant code](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/storage/StorageManagerUtil.java#L68) .

Please let me know if there're any other questions.